From 520d573260cf80ee79b1a9b8700d6116343a9453 Mon Sep 17 00:00:00 2001
From: Dmitry Klochkov <dmitry.klochkov@bell-sw.com>
Date: Wed, 3 Sep 2025 16:05:34 +0000
Subject: [PATCH] tar: add an init_unicode() call

printable_string() that is now used to mask all malicious characters
hides also all unicode characters.  In order to let tar print unicode
characters again, add an init_unicode() call.

Fixes: b09a137dd ("fix 16018 masking of potentially malicious tar/cpio content")
---
 archival/tar.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/archival/tar.c b/archival/tar.c
index 38906ba02..3135282d3 100644
--- a/archival/tar.c
+++ b/archival/tar.c
@@ -119,6 +119,7 @@
 #include "libbb.h"
 #include "common_bufsiz.h"
 #include "bb_archive.h"
+#include "unicode.h"
 /* FIXME: Stop using this non-standard feature */
 #ifndef FNM_LEADING_DIR
 # define FNM_LEADING_DIR 0
@@ -940,6 +941,7 @@ int tar_main(int argc UNUSED_PARAM, char **argv)
 	llist_t *excludes = NULL;
 #endif
 	INIT_G();
+	init_unicode();
 
 	/* Initialise default values */
 	tar_handle = init_handle();
-- 
2.49.0

